src/http/routes/object/getSignedUploadURL.ts

```diff
@@ -0,0 +1,69 @@
+import { FastifyInstance } from 'fastify'
+import { FromSchema } from 'json-schema-to-ts'
+import { createDefaultSchema } from '../../generic-routes'
+import { AuthenticatedRequest } from '../../request'
+
+const getSignedUploadURLParamsSchema = {
+  type: 'object',
+  properties: {
+    bucketName: { type: 'string', examples: ['avatars'] },
+    '*': { type: 'string', examples: ['folder/cat.png'] },
+  },
+  required: ['bucketName', '*'],
+} as const
+const getSignedUploadURLBodySchema = {
+  type: 'object',
+  properties: {
+    expiresIn: { type: 'integer', minimum: 1, examples: [60000] },
+  },
+  required: ['expiresIn'],
+} as const
+
+const successResponseSchema = {
+  type: 'object',
+  properties: {
+    url: {
+      type: 'string',
+      examples: [
+        '/object/sign/upload/avatars/folder/cat.png?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1cmwiOiJhdmF0YXJzL2ZvbGRlci9jYXQucG5nIiwiaWF0IjoxNjE3NzI2MjczLCJleHAiOjE2MTc3MjcyNzN9.s7Gt8ME80iREVxPhH01ZNv8oUn4XtaWsmiQ5csiUHn4',
+      ],
+    },
+  },
+  required: ['url'],
+}
+interface getSignedURLRequestInterface extends AuthenticatedRequest {
+  Params: FromSchema<typeof getSignedUploadURLParamsSchema>
+  Body: FromSchema<typeof getSignedUploadURLBodySchema>
+}
+
+export default async function routes(fastify: FastifyInstance) {
+  const summary = 'Generate a presigned url to upload an object'
+
+  const schema = createDefaultSchema(successResponseSchema, {
+    body: getSignedUploadURLBodySchema,
+    params: getSignedUploadURLParamsSchema,
+    summary,
+    tags: ['object'],
+  })
+
+  fastify.post<getSignedURLRequestInterface>(
+    '/upload/sign/:bucketName/*',
+    {
+      schema,
+    },
+    async (request, response) => {
+      const { bucketName } = request.params
+      const objectName = request.params['*']
+      const { expiresIn } = request.body
```

Comment:
As mentioned in the comment below, we should not accept the `expiresIn` value from the client. We can promote this value as an environment variable with the default of 60s 

That is probably what i would run in production